src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -111,6 +111,13 @@ public override BoundNode VisitUserDefinedConditionalLogicalOperator(BoundUserDe
 
         public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryOperator? applyParentUnaryOperator)
         {
+            if (node.InterpolatedStringHandlerData is InterpolatedStringHandlerData data)
```

Comment:
Aren't we putting a conversion on such binary operators? I would have expected the handling in `LocalRewriter.VisitConversion` would have been sufficient (it handled a conversion on top of a `BoundBinaryOperator` already)